### DISCUSSION TITLE
What should be the behavior when two separate components use the same `x-data="name"`?

### DISCUSSION THREAD

**Discussion by ernstki** [alpinejs/alpine#3031] [1. Help] [answered]
I can't tell if the [Alpine.data](https://alpinejs.dev/globals/alpine-data) documentation is just unclear about this, or if my comprehension of `x-data` is poor, or if what I'm doing is an antipattern, or if I've found a bug.

Minimal example (https://codepen.io/ernstki/pen/BarzgGV):

```html
<section id="one" x-data="form">
  <form @submit.prevent="submitForm">
    <input type="text" x-model="text">
    <textarea x-model="text">
    <input type="sumbit">
    <input type="reset">
  </form>
</section>

<!-- potentially lots of other page content -->

<section id="two" x-data="form">
  <p>text: <span x-text="text"></span>
  <p>textarea: <span x-text="textarea"></span>
  <pre x-text="result"></pre>
</section>

<script>
  document.addEventListener('alpine:init', () => {
    Alpine.data('form', () => ({
      text: "",
      textarea: "",
      result: "",
      submitForm() {
        console.log(this.text);
        console.log(this.textarea);
        this.result = this.text + "\n\n" + this.textarea;
      }
    }))
  })
</script>
```

It _seems_ right to select the smallest scope possible for `x-data`, like on the parent `<div>` or `<section>` of the components that need reactivity. And the [State § Re-usable data](https://alpinejs.dev/essentials/state#re-usable-data) section of the docs has an example which suggests this should work.

> Now that you've registered the "dropdown" data, you can use it inside your markup in as many places as you like:
> 
> ```html
> <div x-data="dropdown">
>     <button @click="toggle">Expand</button>
>  
>     <span x-show="open">Content...</span>
> </div>
>  
> <div x-data="dropdown">
>     <button @click="toggle">Expand</button>
>  
>     <span x-show="open">Some Other Content...</span>
> </div>
> ```

However, written as in the first example above, the components in section two aren't reactive to changes in section one. Written with `x-data="form"` in the `<body>` tag instead of the section tags, all works as expected.

<s>Unrelated, but I also noticed that when the "Reset" button is pressed, the Alpine.data attributes aren't updated (to be blank). Is that a bug, or is this just something Alpine can't check for, so we need to work around it with `@reset` on the `<form>` or something like that?</s> **Edit:** See [below](https://github.com/alpinejs/alpine/discussions/3031#discussioncomment-3125576).

**Comment by ernstki** [comment, marked as answer]
Related to the "Reset" button thing, there's this comment in #686 from @SimoTod:

> If you are using a reactive framework like alpine, you are overriding the standard dom. Your default value needs to be set on your alpine properties used by x-model otherwise when Alpine kicks in, it will override your default value. **The reset thing is a nice to have but not supported at the minute** so you need to add an x-click to reset your models.

So I guess that settles that.

As for why changes to the properties (the `x-model`'d page elements) in one component don't translate to the other component, see @SimoTod's explanation below. Stated a different way, section one and section two's data are separate and encapsulated, and reactive changes don't travel across that boundary. So the behavior see in the Codepen example _is_ the expected behavior.

Thanks, @SimoTod!

**Reply by SimoTod** [reply]
You also misunderstood Alpine.data. It's a factory, every component gets a separate scope with the same variable names but they are not shared.
See Alpine.store for sharing data between components

**Reply by ernstki** [reply]
Ah, thanks. I get it now. Boy, that's a mindset shift though.

Is this obvious enough to go without mention in the docs, or would you consider merging a PR adding a small explanation to that effect, in the places where it would've been helpful to somebody like me, not having a strong background in Vue or any other reactive frameworks?

**Reply by SimoTod** [reply]
It's not my project but the maintainer usually accepts PR for the documentation.
